packages/mobile/src/harmony-native/components/Button/Button/Button.tsx

```diff
@@ -0,0 +1,266 @@
+import type { ReactNativeStyle } from '@emotion/native'
+import type { TextStyle } from 'react-native/types'
+
+import type { IconProps } from 'app/harmony-native/icons'
+
+import { useTheme } from '../../../foundations/theme'
+import { BaseButton } from '../BaseButton/BaseButton'
+import type { ButtonProps } from '../types'
+import { ButtonSize, ButtonType } from '../types'
+
+const clamp = (val: number, min: number, max: number) => {
+  return Math.max(min, Math.min(val, max))
+}
+
+const adjustHexColor = (color: string, percent: number) => {
+  const mod = (100 + percent) / 100
+  let R = parseInt(color.slice(1, 3), 16)
+  let G = parseInt(color.slice(3, 5), 16)
+  let B = parseInt(color.slice(5, 7), 16)
+
+  R = clamp(Math.round(R * mod), 0, 255)
+  G = clamp(Math.round(G * mod), 0, 255)
+  B = clamp(Math.round(B * mod), 0, 255)
+
+  const RR = `${R < 16 ? '0' : ''}${R.toString(16)}`
+  const GG = `${G < 16 ? '0' : ''}${G.toString(16)}`
+  const BB = `${B < 16 ? '0' : ''}${B.toString(16)}`
+
+  return `#${RR}${GG}${BB}`
+}
+
+export const Button = (props: ButtonProps) => {
+  const {
+    color,
+    hexColor,
+    variant = ButtonType.PRIMARY,
+    size = ButtonSize.DEFAULT,
+    disabled,
+    ...baseProps
+  } = props
+  const { isLoading } = baseProps
+
+  const isDisabled = disabled || isLoading
+  const {
+    type,
+    color: themeColors,
+    cornerRadius,
+    shadows,
+    spacing,
+    typography
+  } = useTheme()
+
+  // - Size Styles -
+  const smallStyles: ReactNativeStyle = {
+    gap: spacing.xs,
+    height: spacing.unit8,
+    paddingHorizontal: spacing.m
+  }
+  const smallTextStyles: TextStyle = {
+    fontSize: typography.size.s,
+    fontWeight: `${typography.weight.bold}` as TextStyle['fontWeight'],
+    lineHeight: typography.lineHeight.s,
+    textTransform: 'capitalize'
+  }
+
+  const defaultStyles: ReactNativeStyle = {
+    gap: spacing.s,
+    height: spacing.unit12,
+    paddingHorizontal: spacing.xl
+  }
+  const defaultTextStyles: TextStyle = {
+    fontSize: typography.size.l,
+    fontWeight: `${typography.weight.bold}` as TextStyle['fontWeight'],
+    lineHeight: typography.lineHeight.l,
+    textTransform: 'capitalize'
+  }
+
+  const largeStyles: ReactNativeStyle = {
+    gap: spacing.s,
+    height: spacing.unit16,
+    paddingHorizontal: spacing.xl
+  }
+  const largeTextStyles: TextStyle = {
+    fontSize: typography.size.xl,
+    fontWeight: `${typography.weight.bold}` as TextStyle['fontWeight'],
+    lineHeight: typography.lineHeight.l,
+    letterSpacing: 0.25,
+    textTransform: 'uppercase'
+  }
+
+  // - Variant Styles -
+  const primaryOverrideColor =
+    hexColor ?? (color ? themeColors.special[color] : null)
+
+  const primaryStyles: ReactNativeStyle = {
+    ...(isDisabled && {
+      backgroundColor: themeColors.neutral.n150
+    })
+  }
+  const primaryDynamicStyles = {
```

Comment:
I wouldn't call these "styles" I guess... what do we call these in base button in web?

Reply:
yea styles isn't 100% correct, but they're not really props either. and we don't do this in web bc it's easier to animate things in web